### PR TITLE
[Merged by Bors] - feat: port Analysis.Normed.Ring.Seminorm

### PR DESCRIPTION
**Pull request by AlexKontorovich**
~~This is currently stuck on an issue as indicated in line 111.~~ edit: fixed by the `etaExperiment` option

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Normed/Ring/Seminorm.lean [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+  MonoidWithZeroHom R ℝ
+#align mul_ring_seminorm MulRingSeminorm
+
+/-- A multiplicative norm on a ring `R` is a multiplicative ring seminorm such that `f x = 0`
+implies `x = 0`. -/
+structure MulRingNorm (R : Type _) [NonAssocRing R] extends MulRingSeminorm R, AddGroupNorm R
+#align mul_ring_norm MulRingNorm
+
+attribute [nolint docBlame]
+  RingSeminorm.toAddGroupSeminorm RingNorm.toAddGroupNorm RingNorm.toRingSeminorm MulRingSeminorm.toAddGroupSeminorm MulRingSeminorm.toMonoidWithZeroHom MulRingNorm.toAddGroupNorm MulRingNorm.toMulRingSeminorm
```
Comment (AlexKontorovich) [inline Mathlib/Analysis/Normed/Ring/Seminorm.lean:72]:
```suggestion
  RingSeminorm.toAddGroupSeminorm RingNorm.toAddGroupNorm RingNorm.toRingSeminorm 
    MulRingSeminorm.toAddGroupSeminorm MulRingSeminorm.toMonoidWithZeroHom 
    MulRingNorm.toAddGroupNorm MulRingNorm.toMulRingSeminorm
```

File: Mathlib/Analysis/Normed/Ring/Seminorm.lean [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+  ⟨{ (1 : AddGroupSeminorm R) with
+      mul_le' := fun x y => by
+        by_cases h : x * y = 0
+        ·
+          refine' (if_pos h).trans_le (mul_nonneg _ _) <;>
+            · change _ ≤ ite _ _ _
+              split_ifs
+              exacts[le_rfl, zero_le_one]
+        · change ite _ _ _ ≤ ite _ _ _ * ite _ _ _
+          simp only [if_false, h, left_ne_zero_of_mul h, right_ne_zero_of_mul h, mul_one, le_refl] }⟩
```
Comment (AlexKontorovich) [inline Mathlib/Analysis/Normed/Ring/Seminorm.lean:129]:
```suggestion
          simp only [if_false, h, left_ne_zero_of_mul h, right_ne_zero_of_mul h, mul_one, 
            le_refl] }⟩
```

**Comment by AlexKontorovich** [comment]
Lint is complaining about some `simp` lemmas about `toFun` not being in simp normal form. But when I change that, it's complaining that the lemma is a tautology. So what's the right formulation here? Thanks!

**Comment by int-y1** [comment]
> Lint is complaining about some `simp` lemmas about `toFun` not being in simp normal form. But when I change that, it's complaining that the lemma is a tautology. So what's the right formulation here? Thanks!

I saw 4 lint errors for 4 different theorems. You may have confused them.
```
RingSeminorm.toFun_eq_coe
MulRingSeminorm.toFun_eq_coe
RingNorm.toFun_eq_coe
MulRingNorm.toFun_eq_coe
```

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**